plenum/test/test_node.py

```diff
@@ -379,7 +379,7 @@ def __init__(self,
         self.pluginPaths = pluginPaths
 
         self.testNodeClass = testNodeClass
-        self.nodes = OrderedDict()  # type: Dict[str, TestNode]
+        self.nodes = SortedDict()  # type: Dict[str, TestNode]
```

Comment:
Nodes sort their remotes by **rank** and **not name**. For old style nodes, rank is generated on the basis of name and for new style nodes, it's generated on the basis of order in ledger
You can use `nodes_by_rank` and it will give you correct node for both old style and new style nodes.